app/managers/performance_metrics_manager/index.test.ts

```diff
@@ -47,30 +63,70 @@ describe('load metrics', () => {
             'setImmediate',
             'setInterval',
         ]}).setSystemTime(new Date(TEST_EPOCH));
+        PerformanceMetricsManager = new PerformanceMetricsManagerClass();
     });
     afterEach(async () => {
         jest.useRealTimers();
         NetworkManager.invalidateClient(serverUrl);
         await TestHelper.tearDown();
+        performance.clearMarks();
     });
 
     it('only load on target', async () => {
         performance.mark('nativeLaunchStart');
+        const measure = getMeasure(TEST_EPOCH + INTERVAL_TIME, INTERVAL_TIME);
         const expectedRequest = getBaseReportRequest(measure.timestamp, measure.timestamp + 1);
         expectedRequest.body.histograms = [measure];
 
         PerformanceMetricsManager.setLoadTarget('CHANNEL');
         PerformanceMetricsManager.finishLoad('HOME', serverUrl);
         await TestHelper.tick();
-        jest.advanceTimersByTime(61000);
+        jest.advanceTimersByTime(INTERVAL_TIME);
         await TestHelper.tick();
         expect(mockApiClient.post).not.toHaveBeenCalled();
         PerformanceMetricsManager.finishLoad('CHANNEL', serverUrl);
         await TestHelper.tick();
-        jest.advanceTimersByTime(61000);
+        jest.advanceTimersByTime(INTERVAL_TIME);
+        await TestHelper.tick();
+        expect(mockApiClient.post).toHaveBeenCalledWith(expectedUrl, expectedRequest);
+    });
+
+    it('retry if the mark is not yet present', async () => {
+        const measure = getMeasure(TEST_EPOCH + (RETRY_TIME * 2), RETRY_TIME);
+        const expectedRequest = getBaseReportRequest(measure.timestamp, measure.timestamp + 1);
+        expectedRequest.body.histograms = [measure];
+
+        PerformanceMetricsManager.setLoadTarget('CHANNEL');
+        PerformanceMetricsManager.finishLoad('CHANNEL', serverUrl);
+        await TestHelper.tick();
+        jest.advanceTimersByTime(RETRY_TIME);
+        await TestHelper.tick();
+        performance.mark('nativeLaunchStart');
+        await TestHelper.tick();
+        jest.advanceTimersByTime(RETRY_TIME);
+        await TestHelper.tick();
+        await TestHelper.tick();
```

Comment:
I'm still trying to wrap my head around the tick() function, but in this instance, why is it necessary to have 2 TestHelper.tick()(s)?

Reply:
The tick what is doing is "cleaning the event loop". Some functions that are executed are async, so even after we advance the timers, we have to await for them to be executed. And if an async function calls another async function, we have to wait again.

It is not the prettiest, but works.